tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/aux/CherryPick.scala

```diff
@@ -833,7 +833,7 @@ class CherryPick(rewriter: SymbStateRewriter) {
         val domFromPickedRelation = nextState.arena.topCell
         val domElemsFromPickedRelation = nextState.arena.getHas(pickedRelation).map(e => nextState.arena.getHas(e).head)
         nextState = nextState.updateArena(_.appendHas(domFromPickedRelation, domElemsFromPickedRelation: _*))
-        rewriter.lazyEq.cacheEqConstraints(nextState, Seq((pickedDom, domFromPickedRelation)))
+        nextState = rewriter.lazyEq.cacheEqConstraints(nextState, Seq((pickedDom, domFromPickedRelation)))
```

Comment:
See @konnov , this is what happens when you condone `var nextState`.

Reply:
sure, it could happen with a reference to a wrong `val` too?